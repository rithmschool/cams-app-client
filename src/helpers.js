export const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'

export const ensureCorrectUser = (pathId) => {
  let token = localStorage.getItem('token');
  if(!token){return false;}
  let tokenId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
  return +tokenId === +pathId;
}

export const userID = () => JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

export const config = {
  headers: {
    'Accept':'application/json',
    'ContentType':'application/json',
    'Authorization':'bearer ' + localStorage.getItem('token')
  }
}
