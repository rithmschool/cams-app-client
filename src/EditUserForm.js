import React, {PropTypes, Component} from 'react';
import {BASE_URL, userID, config} from './helpers.js';
import axios from 'axios';

class EditUserForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      current_password: '',
      new_password: '',
      confirm_new_password: '',
      message: ''
    }
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  static contextTypes = {
    router: PropTypes.object
  }

  componentWillMount(){
    axios.get(`${BASE_URL}/api/users/${userID()}`, config())
    .then(response => {
      this.setState({email: response.data.email})
    })
  }

  editUser(config, thisArg) {
    axios.patch(`${BASE_URL}/api/users/${userID()}`, {
      email: thisArg.state.email,
      current_password: this.state.current_password,
      new_password: this.state.new_password,
      confirm_new_password: this.state.confirm_new_password
    }, config)
    .then(response => {
      this.setState({
        message: 'You have successfully updated your account!',
        current_password: '',
        new_password: '',
        confirm_new_password: '',
        email: response.data.email
      })
    }).catch(error => {
      this.setState({
        message: `Please re-enter the correct current password 
          and check to see if the new and confirm password fields are the same.`,
        current_password: '',
        new_password: '',
        confirm_new_password: ''
      })
    })
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit(e) {
    e.preventDefault()
    this.editUser(config(), this)
  }

  render () {
    let message = this.state.message ?
      <p>{this.state.message}</p> :
      null;
    return (
      <div>
        <div className="banner-text">
          <h1 className="banner-bold">Update</h1>
        </div>
        <div className="content">
          <h1>Profile</h1>
          <form onSubmit={this.handleSubmit}>
            <input
              type="email"
              name="email"
              placeholder="new email"
              value={this.state.email}
              onChange={this.handleChange}
            />
            <input
              type="password"
              name="current_password"
              placeholder="current password"
              value={this.state.current_password}
              onChange={this.handleChange}
            />
            <input
              type="password"
              name="new_password"
              placeholder="new password"
              value={this.state.new_password}
              onChange={this.handleChange}
            />
            <input
              type="password"
              name="confirm_new_password"
              placeholder="confirm new password"
              value={this.state.confirm_new_password}
              onChange={this.handleChange}
            />
            <button className="button button-hover" type="submit" value="Submit">Update</button>
            {message}
          </form>
      </div>
    </div>
    )
  }
}

export default EditUserForm;
