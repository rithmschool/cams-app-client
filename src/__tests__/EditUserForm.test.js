import EditUserForm from "../EditUserForm";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import LocalStorageMock from "../setupTests/LocalStorageMock";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { userID } from "../helpers";
import { BASE_URL } from "../helpers";

var mock = new MockAdapter(axios);

beforeAll(() => {
  localStorage.setItem("token", ".eyJpZCI6NH0=.");
  mock
    .onGet(`${BASE_URL}/api/users/${userID()}`)
    .reply(200, { email: "email" });
});

test("EditUserForm snapshot", () => {
  const editUserForm = shallow(<EditUserForm />);
  editUserForm.setState({
    email: "something",
    current_password: "something",
    new_password: "somethingelse",
    confirm_new_password: "somethingelse",
    message: "nothing"
  });
  expect(toJson(editUserForm)).toMatchSnapshot();
});

afterAll(() => {
  mock.restore();
});
