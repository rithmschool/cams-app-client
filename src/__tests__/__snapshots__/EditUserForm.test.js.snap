// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserForm snapshot 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Update
    </h1>
  </div>
  <div
    className="content"
  >
    <h1>
      Profile
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="new email"
        type="email"
        value="something"
      />
      <input
        name="current_password"
        onChange={[Function]}
        placeholder="current password"
        type="password"
        value="something"
      />
      <input
        name="new_password"
        onChange={[Function]}
        placeholder="new password"
        type="password"
        value="somethingelse"
      />
      <input
        name="confirm_new_password"
        onChange={[Function]}
        placeholder="confirm new password"
        type="password"
        value="somethingelse"
      />
      <button
        className="button button-hover"
        type="submit"
        value="Submit"
      >
        Update
      </button>
      <p>
        nothing
      </p>
    </form>
  </div>
</div>
`;
